### PR TITLE
[Wait for #1358] [Pooling] Apply padding property

### PR DESCRIPTION
**Pull request by zhoonit**
**[Pending commit: #1358]**

- [Pooling] Apply padding property

```
This patch applies padding property with tests

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

```

### CODE REVIEW

**Review by jijoongmoon — APPROVED** [review]
Great. ! Now NNTrainer support same & valid.!

File: nntrainer/layers/pooling2d_layer.cpp [hunk 1]
```diff
@@ -65,8 +73,8 @@ int Pooling2DLayer::initialize(Manager &manager) {
     }
   }
 
-  unsigned int eff_in_height = in_dim.height() + padding[0] * 2;
-  unsigned int eff_in_width = in_dim.width() + padding[1] * 2;
+  unsigned int eff_in_height = in_dim.height() + pt + pb;
```
Comment (kparichay) [inline nntrainer/layers/pooling2d_layer.cpp:76]: similar inline functions to conv2d can be helpful here as well.

**Review by kparichay — APPROVED** [review]
LGTM other than the minor coment
Good Job

File: nntrainer/layers/pooling2d_layer.cpp (hunk 1 above)
Comment (zhoonit) [reply nntrainer/layers/pooling2d_layer.cpp:76]: I am thinking to merge some utils in to images (also for preparing thinking extension to conv1d, padding1d).

Let me delay this until merging into v2 :)

Comment (kparichay) [reply nntrainer/layers/pooling2d_layer.cpp:76]: alright.

**Pull request merged**